### PR TITLE
Orderform cancer organism

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
**How to install:**
0.1: clone production cg database to cg-stage on clinical-db `sh dbcopy-prod-to-stage.sh cg`
0.2: update cg-stage database with `cg/scripts/make_panels_optional_in_family.sql`
0.3: update cg-stage database with `cg/scripts/add-data_analysis-to-samples.sql`
0.4: install branch on clinical-db clinical-api: `sh ./update-clinical-api-stage.sh orderform-cancer-organism `
0.5: install branch on clinical-db cgweb: `sh ./update-cgweb-stage.sh orderform-cancer`
0.6: install branch on rasta cg: `sh ./update-cg-stage.sh orderform-cancer-organism `

**How to test:**
Submit Orderforms in the Order Portal (with "#NNNNNN" to avoid creating tickets)

- [ ] 1508 with data analysis fastq 
- [ ] 1508 with data analysis MIP 
- [ ] 1508 with data analysis Balsamic 
- [ ] 1508 with data analysis MIP+Balsamic
- [ ] 1541 external 
- [ ] 1603 microbial 
- [ ] 1604 rml 
- [ ] 1605 metagenome

**Expected outcome:**
Samples created in status-db and LIMS for all type of orderforms

### CODE REVIEW

**Comment by patrikgrenfeldt** [comment]
@ingkebil could you make a code review? :)

File: scripts/add-capture_kit-table.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE `organism` (
```
Comment (ingkebil) [inline scripts/add-capture_kit-table.sql]: Looks like the wrong content for capture kit table.

Could be removed?

File: tests/meta/orders/test_meta_orders_status.py [hunk 2]
```diff
@@ -352,10 +395,102 @@ def test_store_external_bad_apptag(orders_api, base_store, external_status_data)
...
+    # GIVEN a basic store with no samples and a cancer order
+    assert base_store.samples().first() is None
+    assert base_store.families().first() is None
+
+    # WHEN storing the order
+    new_families = orders_api.store_families(
+        customer=balsamic_status_data['customer'],
+        order=balsamic_status_data['order'],
+        ordered=dt.datetime.now(),
+        ticket=1234567,
```
Comment (ingkebil) [inline tests/meta/orders/test_meta_orders_status.py]: ticket numbers are 6 digits only

File: cg/apps/lims/orderform.py [hunk 3]
```diff
@@ -27,6 +27,22 @@
...
+    '1508:14',      # Orderform MIP, Balsamic, sequencing only
+    '1541:6',       # Orderform Externally sequenced samples
+    '1603:6',       # Microbial WGS
+    '1604:8',       # Orderform Ready made libraries (RML)
+    '1605:4',       # Microbial metagenomes
+]
+
+
+def check_orderform_version(document_title):
+    """Raise an error if the orderform is to new or to old for the order portal."""
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/apps/lims/orderform.py]: fixed

File: cg/meta/orders/api.py [hunk 4]
```diff
@@ -142,12 +142,34 @@ def submit_metagenome(self, data: dict) -> dict:
 
     def submit_external(self, data: dict) -> dict:
         """Submit a batch of externally sequenced samples for analysis."""
-        result = self.process_analysis_samples(data)
+        result = self.process_family_samples(data)
         return result
 
-    def submit_scout(self, data: dict) -> dict:
+    def submit_mip(self, data: dict) -> dict:
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/meta/orders/api.py]: refactored to use the same method

File: cg/meta/orders/status.py [hunk 5]
```diff
@@ -136,7 +140,8 @@ def families_to_status(cls, data: dict) -> dict:
             if len(values) > 1:
                 raise ValueError(f"different 'priority' values: {family_name} - {values}")
             priority = values.pop()
-            panels = set(panel for sample in family_samples for panel in sample['panels'])
+            panels = set(panel for sample in family_samples for panel in sample.get('panels',
+                                                                                    set()))
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/meta/orders/status.py]: I didn't manage write this in a more readable way

File: cg/meta/orders/status.py [hunk 6]
```diff
@@ -145,8 +150,9 @@ def families_to_status(cls, data: dict) -> dict:
                     'internal_id': sample.get('internal_id'),
                     'name': sample['name'],
                     'application': sample['application'],
+                    'data_analysis': sample.get('data_analysis'),
                     'sex': sample['sex'],
-                    'status': sample['status'],
+                    'status': sample.get('status','unknown'),
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/meta/orders/status.py]: done

File: cg/store/api/status.py [hunk 7]
```diff
@@ -74,7 +74,9 @@ def families_to_analyze(self, limit: int=50):
                     and_(
                         models.Sample.sequenced_at != None,
                         models.Analysis.completed_at == None,
+                        models.Analysis.completed_at == None,
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/store/api/status.py]: removed

File: scripts/add-capture_kit-table.sql (hunk 1 above)
Comment (patrikgrenfeldt) [reply scripts/add-capture_kit-table.sql]: removed

File: cg/store/api/status.py [hunk 8]
```diff
@@ -74,7 +74,9 @@ def families_to_analyze(self, limit: int=50):
                     and_(
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/store/api/status.py]: renamed to families_to_mip_analyze

File: tests/meta/orders/test_meta_orders_status.py (hunk 2 above)
Comment (patrikgrenfeldt) [reply tests/meta/orders/test_meta_orders_status.py]: removed last digit

**Pull request merged**